### PR TITLE
(SERVER-2095) Remove JRuby 1.7 support

### PR DESCRIPTION
**Pull request by pcarlisle**

### CODE REVIEW

**Comment by haus** [comment]
Have you considered pulling the jruby jar back into the uberjar? It would make using AOT a lot easier for puppetserver.

**Comment by pcarlisle** [comment]
Yeah, but I think I'm going to put it off.

File: project.clj [hunk 1]
```diff
@@ -203,7 +201,7 @@
                                                ;; https://github.com/technomancy/leiningen/issues/2216
                                                ;; Hopefully we can remove those `nil`s (if we care)
                                                ;; and this comment when lein 2.7.2 is available.
-                                               ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
```
Comment (justinstoller) [inline project.clj]: looks like there's a few whitespace oddities in this file (this and the two changes above)

**Comment by pcarlisle** [comment]
still a little broken

**Comment by pcarlisle** [comment]
should be good to go

File: acceptance/README.md [hunk 2]
```diff
@@ -51,7 +51,13 @@ changes.
 
 ### PACKAGE_BUILD_VERSION when testing a new build with local changes
 
-1. Run `lein with-profile ezbake ezbake build`. This will take a while, and
+1. Run
+   ```
+   lein clean
```
Comment (lindboe) [inline acceptance/README.md]: this is applicable to older branches too, right?

Comment (pcarlisle) [reply acceptance/README.md]: Probably yes

Comment (lindboe) [reply acceptance/README.md]: maybe should have separate PR for that

Comment (pcarlisle) [reply acceptance/README.md]: sure we can backport this later

**Comment by lindboe** [comment]
Do we want to deprecate the JRUBY_JAR setting and document it as such (and then possibly also log a warning if it's used?)

**Comment by pcarlisle** [comment]
Right now it's still usable if you want to try a different version. If we do SERVER-2157 then we should do that.

**Comment by lindboe** [comment]
I was just thinking about not supporting an option for longer than we need to. SERVER-2157 meaning we should still support it or no?

**Comment by lindboe** [comment]
talked in person: Patrick's adding a note to SERVER-2157 that we should remove the JRUBY_JAR support then and log a warning because it won't do anything meaningful anymore

File: test/unit/puppetlabs/services/jruby/jruby_puppet_core_test.clj [hunk 3]
```diff
@@ -167,10 +167,7 @@
                                     nil)]
 
       (testing "jruby-config default values are used if not provided"
-        ; jruby-utils defaults to jit when running 9k
-        (if jruby-schemas/using-jruby-9k?
-          (is (= :jit (:compile-mode initialized-jruby-config)))
-          (is (= :off (:compile-mode initialized-jruby-config))))
+        (is (= :jit (:compile-mode initialized-jruby-config)))
```
Comment (Magisus) [inline test/unit/puppetlabs/services/jruby/jruby_puppet_core_test.clj]: Do we plan to update jruby-utils to remove the logic around selecting the default compile-mode? https://github.com/puppetlabs/jruby-utils/blob/6a77927e73c9fcade6265f8044242224565f1ace/src/clj/puppetlabs/services/jruby_pool_manager/jruby_core.clj#L24-L29

Comment (lindboe) [reply test/unit/puppetlabs/services/jruby/jruby_puppet_core_test.clj]: Oh yeah, good call, that will need an update.

Comment (Magisus) [reply test/unit/puppetlabs/services/jruby/jruby_puppet_core_test.clj]: I'll file a ticket about that.

**Pull request merged**